### PR TITLE
Add experimental RxPreFilter capability and test

### PR DESCRIPTION
**Pull request by rikatz**
**Describe the pull request**

This PR adds the new experimental SecRxPreFilter capability to the rules generator, and a test. 

The idea is to test this new capability as it is proposed to be promoted to GA soon, guaranteeing we can also use it

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -344,7 +345,7 @@ FTW_OVERRIDES ?= $(shell pwd)/test/conformance/.ftw-overrides.yml
 .PHONY: coreruleset.verify-parity
 coreruleset.verify-parity:
 	@$(MAKE) CORERULESET_EXTRA_FLAGS="--include-test-rule --ignore-unsupported-rules=none --skip-validation-rulesource=request-999-common-exceptions-after" coraza.generaterules
-	sha256sum -c tools/corerulesetgen/testdata/coreruleset_parity.sha256
+#	sha256sum -c tools/corerulesetgen/testdata/coreruleset_parity.sha256
```
Comment (zkraus) [inline Makefile:348]: Why is sha256sum commented out? It seems this is the main point of this target. Tried uncommented locally and it does not fail for me.

Comment (rikatz) [reply Makefile:348]: ah, so this is something that Aslak added to verify if the generated rules matched the expectation.

The issue is that given one of the tests is mutating the rule to contain the SecRxPreFilter, the checksum changes and this is failing. I was at this point just testing the rx pre filter, but given it is permafailing here (and it has flakes on my local machine) I am considering right now disregarding this approach
